haystack/tools/parameters_schema_utils.py

```diff
@@ -47,6 +47,42 @@ def _get_param_descriptions(method: Callable) -> Tuple[str, Dict[str, str]]:
     return parsed_doc.short_description or "", param_descriptions
 
 
+def _get_component_param_descriptions(component: Any) -> Tuple[str, Dict[str, str]]:
+    """
+    Get parameter descriptions from a component, handling both regular Components and SuperComponents.
+
+    For regular components, this extracts descriptions from the run method's docstring.
+    For SuperComponents, this extracts descriptions from the underlying pipeline components.
+
+    :param component: The component to extract parameter descriptions from
+    :returns: A tuple of (short_description, param_descriptions)
+    """
+    # Import here to avoid circular imports
+    from haystack.core import SuperComponent
+
+    # Get descriptions from the component's run method
+    short_desc, param_descriptions = _get_param_descriptions(component.run)
+
+    # If it's a SuperComponent, enhance the descriptions from the original components
+    if isinstance(component, SuperComponent):
+        for super_param_name, pipeline_paths in component.input_mapping.items():
```

Comment:
Just a heads-up: `input_mapping` is optional. Parameters will be auto-resolved if possible.

Reply:
Hey @mathislucka - I double checked this - in the latest version of super component we create input_mapping in case user never provided it. So all good, right?

Reply:
Ah ok, all good then!